### PR TITLE
fix: Move all `aten` PRs to Dynamo converter registry

### LINKED ISSUE #2068: Add all open `aten` converter PRs to new branch
- Merge Dynamo converter registry #1942 into branch and then refactor all merged converter implementations to use this registry.

### PR DESCRIPTION
**Pull request by gs-olive**
# Description
**The below are now squashed**

The structure of this PR is as follows:
-  The first commit is a squashed set of 16 PRs which add `aten` improvements and converters, from existing PRs with the label [Story: ATen Op Support](https://github.com/pytorch/TensorRT/pulls?q=is%3Apr+is%3Aopen+label%3A%22Story%3A+ATen+Op+Support%22)
- The second commit **reverts** all of the changes to `py/torch_tensorrt/fx`
- The third commit contains **all** of the changes made to `py/torch_tensorrt/fx`, replayed on `py/torch_tensorrt/dynamo`, with required import, reference, and code updates. This sometimes includes code duplication where absolutely necessary to add some new functionality
- Subsequent commits add various improvements or modifications to these converters

Fixes #2068 

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ - ] I have added tests to verify my fix or my feature
  - Each PR has its own tests
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1056,6 +1072,7 @@ jobs:
       - test-dynamo-compile
       - test-dynamo-compile-core
       - test-dynamo-fx_ts
+      - test-dynamo_converters
```
Comment (narendasan) [inline .circleci/config.yml:1075]: test-dynamo-converters?

File: py/torch_tensorrt/dynamo/converters/aten_ops_converters.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    kwargs: Dict[str, Argument],
+    name: str,
+) -> Union[TRTTensor, Sequence[TRTTensor]]:
+    kwargs_new = {
+        "input": args[0],
+        "other": args[1],
+    }
+    rounding_mode = kwargs.get("rounding_mode")
+    if rounding_mode is None:
+        return acc_ops_converters.acc_ops_div(network, target, None, kwargs_new, name)
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/converters/aten_ops_converters.py:77]: why are we calling out to acc_ops here?

File: py/torch_tensorrt/dynamo/converters/aten_ops_converters.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+@dynamo_tensorrt_converter(torch.ops.aten.gelu.default)
+def aten_ops_gelu(
+    network: TRTNetwork,
+    target: Target,
+    args: Tuple[Argument, ...],
+    kwargs: Dict[str, Argument],
+    name: str,
+) -> Union[TRTTensor, Sequence[TRTTensor]]:
+    return activation.gelu(
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/converters/aten_ops_converters.py:110]: Will this work?

File: py/torch_tensorrt/dynamo/converters/converter_utils.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
+from enum import Enum, auto
+
+from torch_tensorrt.fx.types import (
+    TRTDataType,
+    TRTNetwork,
+    TRTTensor,
+)
+
+
+class SourceIR(Enum):
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/converters/converter_utils.py:10]: Let pull out SourceIR into its own file

File: py/torch_tensorrt/dynamo/converters/impl/activation.py
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/converters/impl/activation.py:1]: Add:
```python
from torch_tensorrt.fx.converters.impl.activation import *
```

File: py/torch_tensorrt/dynamo/converters/impl/__init__.py
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/converters/impl/__init__.py:1]: Add the line:
```python
from torch_tensorrt.fx.converters.impl import convolution
```
Watch collisions, such as `activation.py`. In `dynamo` activation, import `fx` activation.

Comment (gs-olive) [reply py/torch_tensorrt/dynamo/converters/impl/__init__.py:1]: Move all `impl` imports to use module name directly + reference `dynamo` version.

File: py/torch_tensorrt/dynamo/converters/test_utils.py
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/converters/test_utils.py:1]: Options for improved testing:

- Subclass `DispatchTestCase`
- Modify this version directly

File: py/torch_tensorrt/dynamo/converters/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
+import torch
+import logging
+from typing import Callable, List, Optional, Set
+
+import torch_tensorrt.fx.tracer.dispatch_tracer.aten_tracer as aten_tracer
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/converters/test_utils.py:5]: Prototype testing with `torch._export`

File: py/torch_tensorrt/dynamo/converters/aten_ops_converters.py (hunk 3 above)
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/converters/aten_ops_converters.py:110]: Resolved with fixed import

File: py/torch_tensorrt/dynamo/converters/aten_ops_converters.py (hunk 2 above)
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/converters/aten_ops_converters.py:59]: Since there was a change to this operator in the `trunc_div` PR #1869, we need to copy the entire converter to ensure the changes are reflected

**Review by borisfom — APPROVED** [review]

File: py/torch_tensorrt/_Input.py [hunk 6]
```diff
@@ -68,6 +68,17 @@ def __init__(self, *args, **kwargs):
...
+        # Compatibility code for switching over from InputTensorSpec
+        if "shape" in kwargs and "shape_ranges" in kwargs:
+            assert (
+                len(kwargs["shape_ranges"]) == 1 and len(kwargs["shape_ranges"][0]) == 3
+            )
+            del kwargs["shape"]
+
+            kwargs["min_shape"] = kwargs["shape_ranges"][0][0]
+            kwargs["opt_shape"] = kwargs["shape_ranges"][0][1]
+            kwargs["max_shape"] = kwargs["shape_ranges"][0][2]
```
Comment (gs-olive) [inline py/torch_tensorrt/_Input.py:80]: Is it preferable to include this compatibility code in `Input` for user convenience, or to switch over all existing implementations to use `Input`'s schema formally?

**Review by apbose — APPROVED** [review]

File: py/torch_tensorrt/_Input.py (hunk 6 above)
Comment (narendasan) [reply py/torch_tensorrt/_Input.py:80]: We should drop this. Its undocumented and confusing.

**Review by narendasan — APPROVED** [review]
LGTM
